### ISSUE TITLE
" AS " column name assignment not usable in WHERE

### ISSUE THREAD

**Issue by syshex** [orientechnologies/orientdb#3494]
https://groups.google.com/forum/#!topic/orient-database/EI2CEkDV4RI

```
select announcementNumber, count(*) as total from Announcement  group by announcementNumber
```

I get a list of announcementNumber s and another column named total with values ranging from 1 to 5 

```
select announcementNumber, count(*) as total from Announcement where total > 1 group by announcementNumber
```

returns nothing. , but should return something. Problem is "total > 1" is not being evaluated with the value from column "total"

Workaround I used :

```
select *
from (
 select announcementNumber, count(*) as total from Announcement group by announcementNumber 
)
where total > 1 
```

**Comment by syshex** [comment]
using 2.0

**Comment by lvca** [comment]
You can also use the LET to assign a variable and reuse it in WHERE: in OrientDB projections are computed after the where.

**Comment by luigidellaquila** [comment]
In standard SQL you cannot use aliases for aggregate operations in the WHERE condition. To work around this limitations, relational databases provide HAVING keyword.

OrientDB does not support HAVING, so you have to use a subquery for this.

V 2.2 does not support aliases even in ORDER BY, but this limitation was fixed in v 3.0

Closing as invalid

Thanks

Luigi
